File: scratch/query_100_stock.py
import psycopg2
from psycopg2.extras import RealDictCursor

conn_str = "postgresql://zenvix:zenvix_secure_2026!@localhost:5432/zenvix_prod?schema=public"

try:
    conn = psycopg2.connect(conn_str)
    cur = conn.cursor(cursor_factory=RealDictCursor)
    
    # We want to find stock levels and item details for SKU starting with '100'
    query = """
        SELECT s.id as stock_id, s.location_id, l.name as location_name, s.on_hand, s.available, s.product_id, i.sku, i.name as product_name
        FROM stock_levels s
        JOIN item_masters i ON s.product_id = i.id
        JOIN locations l ON s.location_id = l.id
        WHERE i.tenant_id = 'bambu-tenant' AND i.sku LIKE '100%'
        ORDER BY i.sku, l.name;
    """
    cur.execute(query)
    rows = cur.fetchall()
    
    print("--- Stock levels for SKU starting with '100' ---")
    for r in rows:
        print(f"SKU: {r['sku']} | Name: {r['product_name']} | Location: {r['location_name']} | On Hand: {r['on_hand']} | Available: {r['available']}")
        
    cur.close()
    conn.close()
except Exception as e:
    print("Error:", e)

File: scratch/query_100_stock.ts
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient();

async function main() {
  const count = await prisma.item_masters.count();
  console.log("Total items in item_masters:", count);

  if (count > 0) {
    const items = await prisma.item_masters.findMany({
      take: 10
    });
    console.log("First 10 items in DB:");
    for (const i of items) {
      console.log(`ID: ${i.id} | SKU: ${i.sku} | Name: ${i.name} | Tenant: ${i.tenant_id}`);
    }
  }
}

main()
  .catch((e) => console.error(e))
  .finally(async () => {
    await prisma.$disconnect();
  });
